### PR TITLE
Flip Route button + revert to using OverviewViewportState +.show() with route lines over VAMOS network for RoutePreview

### PR DESCRIPTION
**Pull request by jkgz**

### CODE REVIEW

**Comment by jkgz** [comment]
There were already a couple issues with using .showcase() + StaticViewportState that I discovered, but using navigationMapView for preview still has advantages. Now using .show() (for route preview drawing) + other navigationMapView functions to show route durations and waypoints. Working on flip route functionality, it also needed these changes + a change using allowCameraSync to not flash unnecessarily, so I included flip route in this PR.

File: bikestreets-ios/Map/DefaultMapsViewController.swift [hunk 1]
```diff
@@ -43,6 +43,11 @@ final class DefaultMapsViewController: MapsViewController {
     (sheetHeightInspectionView.lastFrameBroadcast?.height ?? 0) + 24
   }
   
+  /// If the camera bottom inset is too small when using overview viewport during route preview, the map zooms way out to accommodate the route geometry. Setting a minimum value avoids this.
+  private var minCameraBottomInsetForPreview: CGFloat {
+    return min(cameraBottomInset, UIScreen.main.bounds.size.height * 0.8)
```
Comment (mattrobmattrob) [inline bikestreets-ios/Map/DefaultMapsViewController.swift:48]: Should this be `max`?

File: bikestreets-ios/Search/DirectionPreviewViewController.swift [hunk 2]
```diff
@@ -146,6 +146,19 @@ final class DirectionPreviewViewController: UIViewController {
       fatalError("Unsupported state")
     }
   }
+  
+  // MARK: Flip Route
+  
+  @objc func flipRouteButtonClicked(_ sender: UIButton) {
```
Comment (mattrobmattrob) [inline bikestreets-ios/Search/DirectionPreviewViewController.swift:152]:
```suggestion
  @objc private func flipRouteButtonClicked(_ sender: UIButton) {
```

**Review by mattrobmattrob — APPROVED** [review]

File: bikestreets-ios/Map/DefaultMapsViewController.swift (hunk 1 above)
Comment (jkgz) [reply bikestreets-ios/Map/DefaultMapsViewController.swift:48]: No, it should be max, but the way I wrote the comment makes it confusing. I'll update the comment to "If the camera bottom inset is too large..." A large indent => large bottom inset. Small indent => small bottom inset.

Comment (jkgz) [reply bikestreets-ios/Map/DefaultMapsViewController.swift:48]: I think it should be min, but I wrote the comment in a confusing way. If the cameraBottomInset is small (small detent), then we want to use it, but if the cameraBottomInset is large (large dentent) and greater than, say, 80% of the screen height, we want the minimum of the two values, in this case, the 80% of screen height.

I will rewrite the comment to be more clear. Agree with my logic?

Comment (mattrobmattrob) [reply bikestreets-ios/Map/DefaultMapsViewController.swift:50]: Yup! Was mostly going on the comment and function name. 👍

**Pull request merged**